File: Bit Manipulation/GeneratingEverySubsetWithBitmask/EverySubset.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int>Vector; int Size;
void Solve(){
    cout << "All possible subsets are:\n";
    int Index = 1;
    for (int i = 0; i < (1 << Size); i++){
        cout << Index << ") ";
        for (int j = 0; j < Size; j++){
            if (i & (1 << j)){
                cout  << Vector[j] << ' ';
            }
        }
        Index++;
        cout << '\n';
    }
}
int main(){
    cout << "Enter size of vector:\n";
    cin >> Size;
    cout << "Enter your vector elements:\n";
    for (int i = 0; i < Size; i++){
        int Element; cin >> Element;
        Vector.push_back(Element);
    }
    Solve();
	return 0;
}

File: Bit Manipulation/GeneratingEverySubsetWithBitmask/java/EverySubset.java
/**
 * Generates every subset of the array.
 * For e.g.
 * array = [1, 2, 3]
 * Outputs:
 * 1
 * 2
 * 1 2 
 * 3
 * 1 3 
 * 2 3
 * 1 2 3
 */   
import java.util.Scanner;

class EverySubset{

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        System.out.println("Input your array size : ");
        int n = scanner.nextInt();

        int[] array = new int[n];
        System.out.println("Please enter the "+n+" elements :");
        for(int i = 0; i < n; i++){
            array[i] = scanner.nextInt();
        }

        System.out.println("All subsets of array are : ");
        allSubsets(array, n);
    }

    private static void allSubsets(int[] array, int n){
        int totalSubsetCoount = (1 << n); //For an array of size n, there can be a total of 2^n subsets.

        for(int i = 0; i < totalSubsetCoount; i++){
            for(int j = 0; j < n; j++) {
                // If the jth bit in the number i is 1,
                // then jth element of array must be included in the present subset.
                if((i & (1 << j)) != 0) {
                    System.out.print(array[j] + " ");
                }
            }

            System.out.println();
        }
    }
}